File: practice-projects/gin-ddd/internal/infrastructure/persistence/user/user_do.go
// Package user 提供用户聚合根的持久化适配。
package user

import (
	"database/sql"
	"time"
)

// UserDO 用户数据对象,所有 db tag 集中于此。
// Phone/Address 用 sql.NullString 适配数据库 NULL 列,与 domain 层使用的
// *string 解耦,转换由 UserConverter 完成。
type UserDO struct {
	ID          int64          `db:"id"`
	Name        string         `db:"name"`
	Email       string         `db:"email"`
	Phone       sql.NullString `db:"phone"`
	Address     sql.NullString `db:"address"`
	CreatedTime time.Time      `db:"created_time"`
	UpdatedTime time.Time      `db:"updated_time"`
}

File: practice-projects/springboot4ddd/src/main/java/com/github/microwind/springboot4ddd/infrastructure/repository/user/UserDO.java
package com.github.microwind.springboot4ddd.infrastructure.repository.user;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 用户数据对象（Data Object）
 *
 * <p>仅承担"与 users 表字段一对一映射"的职责。领域模型 {@code User}
 * 不再背负持久化细节，转换由 {@link UserConverter} 显式完成。
 *
 * <p>当前数据库未包含 {@code wechat} 列，DO 保留该字段供未来扩展，
 * RowMapper 在读取时容错处理。
 *
 * @author jarry
 * @since 1.0.0
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserDO {

    private Long id;
    private String name;
    private String email;
    private String phone;
    private String wechat;
    private String address;
    private LocalDateTime createdTime;
    private LocalDateTime updatedTime;
}
